### PR TITLE
MGMT-10850: Message for Day2 reboot is misleading

### PR DESCRIPTION
**Pull request by paul-maidment**
The install progress message in for a node is misleading when a Day2
host is rebooting as it states that "No further updates will be posted"

This is inaccurate as there are certain activities that could lead to an
update to the agent status. For example CSR auto signature operations.

This commit rectifies this problem by removing the confusing part of the
message.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @omertuc 
/cc @ori-amizur 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by omertuc** [comment]
The message was good in the context of REST AI, it's not good in the context of ZTP. The solution is not to change the message, the solution is to remove it altogether in the context of ZTP

File: internal/host/host.go [hunk 1]
```diff
@@ -586,8 +588,13 @@ func (m *Manager) UpdateInstallProgress(ctx context.Context, h *models.Host, pro
 			swag.StringValue(h.Status), models.HostStatusError, statusInfo)
 	case models.HostStageRebooting:
 		if swag.StringValue(h.Kind) == models.HostKindAddToExistingClusterHost {
+			statusInfoMessage := statusInfoRebootingDay2
+			if m.kubeApiEnabled {
+				statusInfoMessage = statusInfoRebootingDay2ManagedInstallation
```
Comment (omertuc) [inline internal/host/host.go:593]: We don't need any message here, the user doesn't need to do anything, just like they don't need to have a message / do anything for a regular rebooting stage during a normal day-1 installation

File: internal/host/host.go [hunk 2]
```diff
@@ -586,8 +588,13 @@ func (m *Manager) UpdateInstallProgress(ctx context.Context, h *models.Host, pro
 			swag.StringValue(h.Status), models.HostStatusError, statusInfo)
 	case models.HostStageRebooting:
 		if swag.StringValue(h.Kind) == models.HostKindAddToExistingClusterHost {
+			statusInfoMessage := statusInfoRebootingDay2
+			if m.kubeApiEnabled {
+				statusInfoMessage = ""
```
Comment (omertuc) [inline internal/host/host.go:593]: Why is this needed? Just add an `&&` to the `if swag.StringValue(h.Kind) == models.HostKindAddToExistingClusterHost` condition above

Comment (paul-maidment) [reply internal/host/host.go:593]: A few lines below we update the host with a status, this update does not depend on KubeAPI being enabled. 

I don't think it's appropriate to add the condition to the `if` statement for this reason.

File: internal/host/host.go [hunk 3]
```diff
@@ -585,9 +587,11 @@ func (m *Manager) UpdateInstallProgress(ctx context.Context, h *models.Host, pro
 		_, err = hostutil.UpdateHostStatus(ctx, logutil.FromContext(ctx, m.log), m.db, m.eventsHandler, h.InfraEnvID, *h.ID,
 			swag.StringValue(h.Status), models.HostStatusError, statusInfo)
 	case models.HostStageRebooting:
-		if swag.StringValue(h.Kind) == models.HostKindAddToExistingClusterHost {
+		if swag.StringValue(h.Kind) == models.HostKindAddToExistingClusterHost && !m.kubeApiEnabled{
+			statusInfoMessage := statusInfoRebootingDay2
+
```
Comment (omertuc) [inline internal/host/host.go:592]:
```suggestion
```

File: internal/host/host.go [hunk 4]
```diff
@@ -585,9 +587,11 @@ func (m *Manager) UpdateInstallProgress(ctx context.Context, h *models.Host, pro
...
 			_, err = hostutil.UpdateHostProgress(ctx, logutil.FromContext(ctx, m.log), m.db, m.eventsHandler, h.InfraEnvID, *h.ID,
-				swag.StringValue(h.Status), models.HostStatusAddedToExistingCluster, statusInfoRebootingDay2,
+				swag.StringValue(h.Status), models.HostStatusAddedToExistingCluster, statusInfoMessage,
```
Comment (omertuc) [inline internal/host/host.go:594]:
```suggestion
				swag.StringValue(h.Status), models.HostStatusAddedToExistingCluster, statusInfoRebootingDay2,
```

File: internal/host/host.go [hunk 5]
```diff
@@ -585,9 +587,11 @@ func (m *Manager) UpdateInstallProgress(ctx context.Context, h *models.Host, pro
 		_, err = hostutil.UpdateHostStatus(ctx, logutil.FromContext(ctx, m.log), m.db, m.eventsHandler, h.InfraEnvID, *h.ID,
 			swag.StringValue(h.Status), models.HostStatusError, statusInfo)
 	case models.HostStageRebooting:
-		if swag.StringValue(h.Kind) == models.HostKindAddToExistingClusterHost {
+		if swag.StringValue(h.Kind) == models.HostKindAddToExistingClusterHost && !m.kubeApiEnabled{
```
Comment (filanov) [inline internal/host/host.go:590]: Maybe instead of passing another parameter you can check if the host have a kube-key? https://github.com/openshift/assisted-service/blob/7f0a41b671297e987ae07a6e55d32f82a5c0c974/internal/common/db.go#L78

Comment (paul-maidment) [reply internal/host/host.go:590]: I think that `h` here is a models.Host, whereas the KubeKeyNamespace field is an attribute of common.Host. 
Not sure we can use this here?

**Comment by omertuc** [comment]
Feel free to unhold once @paul-maidment and @filanov agree

File: internal/host/host.go (hunk 5 above)
Comment (filanov) [reply internal/host/host.go]: Make sense.
